benchexec/tablegenerator/react-table/src/components/ReactTable.js

```diff
@@ -97,6 +100,50 @@ export default class Table extends React.Component {
     ];
     this.typingTimer = -1;
   }
+
+  componentDidMount() {
+    window.addEventListener("popstate", this.updateTools, false);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("popstate", this.updateTools, false);
+  }
+
+  updateTools = () => {
+    this.setState((prevState) => ({
+      tools: this.adjustToUrlParams(prevState.tools),
+    }));
+  };
+
+  adjustToUrlParams(tools) {
```

Comment:
Because column visibility influences all tabs, this should maybe be in a different file?